src/compiler/crystal/tools/dependencies.cr

```diff
@@ -80,5 +112,72 @@ module Crystal
     private def match_patterns?(patterns, paths)
       patterns.any? { |pattern| paths.any? { |path| File.match?(pattern, path) } }
     end
+
+    class List < DependencyPrinter
+      private def print_file(filename, parent, filter, unseen)
+        @io.print path(filename)
+        if comment = edge_comment(filter, unseen)
+          @io.print " "
+          @io.print comment
+        end
+        @io.puts
+      end
+
+      private def print_indent
+        @io.print "  " * @stack.size unless @stack.empty?
+      end
+
+      private getter? wants_indent : Bool { @format.tree? }
+    end
+
+    class Dot < DependencyPrinter
+      def start_format
+        @io.puts "digraph G {"
+      end
+
+      def end_format
+        @io.puts "}"
+      end
+
+      private def print_file(filename, parent, filter, unseen)
+        return unless parent
+
+        @io.print "  "
+        @io.print path(parent)
+        @io.print " -> "
+        @io.print path(filename)
+        if comment = edge_comment(filter, unseen)
+          @io.print %( [label="#{comment}"])
+        end
+        @io.puts
+      end
+
+      private def path(filename)
+        super.to_s.inspect
+      end
+    end
+
+    class Mermaid < DependencyPrinter
+      def start_format
+        @io.puts "graph LR"
+      end
+
+      def end_format
+      end
+
```

Comment:
Obsolete stub.

```suggestion
```

Reply:
Yeah, I'm a bit torn on this one. There's some merit in explicitly saying that the end is empty. But I suppose it's better to leave is out.